ament_index_cpp/CMakeLists.txt

```diff
@@ -20,14 +20,20 @@ ament_export_libraries(${PROJECT_NAME})
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  set(THREADS_PREFER_PTHREAD_FLAG ON)
+  find_package(Threads REQUIRED)
   # prevent cppcheck from being found, since it fails to process the file utest.cpp
   set(ament_cmake_cppcheck_FOUND TRUE)
   ament_lint_auto_find_test_dependencies()
 
   ament_add_gtest(${PROJECT_NAME}_utest test/utest.cpp)
   if(TARGET ${PROJECT_NAME}_utest)
     target_include_directories(${PROJECT_NAME}_utest PUBLIC include)
-    target_link_libraries(${PROJECT_NAME}_utest ${PROJECT_NAME})
+	if(WIN32)
+   		target_link_libraries(${PROJECT_NAME}_utest ${PROJECT_NAME})
+	else()
+		target_link_libraries(${PROJECT_NAME}_utest ${PROJECT_NAME} Threads::Threads)
+	endif()
```

Comment:
Please don't duplicate common logic. Keep the existing `target_link_libraries` and conditionally call it again for `Threads`.